internal/cli/api/api.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2024 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/iancoleman/strcase"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/api"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/log"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+func Builder() *cobra.Command {
+	apiCmd := createRootAPICommand()
+
+	for _, tag := range api.Commands {
+		tagCmd := createAPICommandGroupToCobraCommand(tag)
```

Comment:
[q] non blocking comment I'm curious on why did you decide to go on a path where you store a custom struct and at runtime generate the cobra commands? wouldn't it be more efficient to generate cobra commands straight away?